templates/partials/_share_and_donate_page.liquid

```diff
@@ -0,0 +1,95 @@
+<div class="header-logo header-logo--light">
+  <a href="{{ 'footer.home_url' | t }}">
+    <div class="header-logo__logo sumofus-logo--positive"></div>
+  </a>
+</div>
+
+<div class="center-content center-content--share-and-donate">
+  <div class="center-content__one-column">
+
+    <div class="center-content__central-square">
+
+      <h1 class="thank-you__thanks">{{ message }}</h1>
+      <div class="thank-you__cta">
+        <span class="two-step__question">{{ 'share.two_step.cta' | t }}</span>
+        <span class="two-step__declined hidden-closed">
+          {% include 'Fundraiser', ref: 'post-action-donate', freestanding: true %}
+        </span>
+        <span class="two-step__accepted hidden-closed">{{ 'share.two_step.accepted' | t }}</span>
+      </div>
+
+      <div class="center-content__centered-element">
+        <div class="two-step__question">
+          <div class="share-buttons">
+            <div class="share-buttons__button button two-step__button two-step__accept">
+              {{ 'share.two_step.accept' | t }}
+            </div>
+            <div class="share-buttons__button button two-step__button two-step__decline">
+              {{ 'share.two_step.decline' | t }}
+            </div>
+          </div>
+        </div>
+        </div>
+        <div class="two-step__accepted hidden-closed">
+          {% include 'Share' %}
+        </div>
+        <div class="two-step__after-donation hidden-closed">
+          <span> Thank you for donating </span>
+        </div>
+      </div>
+    </div>
+
+  </div>
+</div>
+
+<script type="text/javascript">
+  $(document).ready(function(){
+    $('.two-step__accept').on('click', function(){
+      $('.two-step__question').addClass('hidden-closed');
+      $('.two-step__declined').addClass('hidden-closed');
+      $('.two-step__accepted').removeClass('hidden-closed');
+    });
+    $('.two-step__decline').on('click', function(){
+      $('.two-step__question').addClass('hidden-closed');
+      $('.two-step__accepted').addClass('hidden-closed');
+      $('.two-step__declined').removeClass('hidden-closed');
+      if($(window).width() < 701) {
+        revealMobileFundraiser();
+      }
+    });
+
+    $.subscribe('fundraiser:transaction_success', function(event, responseData, formData) {
+      var followUpUrl, redirected;
+
+      followUpUrl = new URI(responseData.follow_up_url || "{{ follow_up_url }}");
+      followUpUrl.addQuery('donated', 'true');
+      redirected = formData.storeInVault && window.champaign.redirectors.RegisterMemberRedirector.attemptRedirect(followUpUrl.toString(), formData.member)
+
+      if(!redirected) {
+        showThankYouMessage();
+      }
+    });
+
+    checkIfAlreadyDonated();
+
+    function checkIfAlreadyDonated() {
+      var regex = /donated/i;
+      if(regex.test(window.location.search.substring(1))) {
```

Comment:
It's checking for the query param `donated`, not the best way to do it. But now that I have the `urijs` library I can probably do it in a nicer way. I'll fix this.